cmd/config/internal/commands/cmdcreatesetter.go

```diff
@@ -73,18 +75,23 @@ func (r *CreateSetterRunner) runE(c *cobra.Command, args []string) error {
 }
 
 func (r *CreateSetterRunner) preRunE(c *cobra.Command, args []string) error {
+	valueSetFromFlag := isFlagSet("value", c)
 	var err error
 	r.Set.SetPartialField.Setter.Name = args[1]
-	r.Set.SetPartialField.Setter.Value = args[2]
 	r.CreateSetter.Name = args[1]
-	r.CreateSetter.FieldValue = args[2]
+	if valueSetFromFlag {
+		r.CreateSetter.FieldValue = r.Set.SetPartialField.Setter.Value
+	} else if len(args) > 2 {
+		r.Set.SetPartialField.Setter.Value = args[2]
+		r.CreateSetter.FieldValue = args[2]
+	}
 	r.CreateSetter.FieldName, err = c.Flags().GetString("field")
 	if err != nil {
 		return err
 	}
 
 	if setterVersion == "" {
-		if len(args) < 3 {
+		if len(args) < 2 || !valueSetFromFlag && len(args) < 3 {
```

Comment:
Don't need to support this flag for the deprecated setter style.

```go
!c.Flag("value").Changed && len(args) < 3
```

Reply:
this seems like it was dead code -- len(args) < 3 wasn't allowed anyway

Reply:
Done.